### PR TITLE
Modular pose extraction

### PR DESCRIPTION
**Pull request by mpiseno**
## Motivation and Context

One piece of feedback I got from several people was that the image extractor does not extract enough images for a single scene. One direction we can take to resolve this is to allow people to create their own custom rules for pose extraction. In this PR I have modularized the PoseExtractor class to subclass different types of pose extraction and support the creating of custom pose extractors. I will also add a section on to the tutorial about how to create a custom pose extractor in a future PR.

relevant issue: https://github.com/facebookresearch/habitat-sim/issues/552#issuecomment-606197026

## How Has This Been Tested

The current tests cover the two main ways of pose extraction already (closest and panorama).

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by dhruvbatra** [comment]
Motivation makes sense to me. I know @arjunmajum already found this useful. 

Let me know when there's a tutorial writeup. I'd like to play around with this then.

**Comment by mpiseno** [comment]
> Motivation makes sense to me. I know @arjunmajum already found this useful.
> 
> Let me know when there's a tutorial writeup. I'd like to play around with this then.

Will do, I'll likely have time to create a detailed tutorial within the next week.

File: habitat_sim/utils/data/data_extractor.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 from habitat_sim.agent import AgentState
 from habitat_sim.utils.common import quat_from_two_vectors
 from habitat_sim.utils.data.data_structures import ExtractorLRUCache
-from habitat_sim.utils.data.pose_extractor import PoseExtractor
+from habitat_sim.utils.data.pose_extractor import *
```
Comment (erikwijmans) [inline habitat_sim/utils/data/data_extractor.py:15]: No `import *`

File: habitat_sim/utils/data/data_extractor.py [hunk 2]
```diff
@@ -245,6 +249,18 @@ def _generate_label_map(self, scene, verbose=False):
 
         return instance_id_to_name
 
+    def _register_pose_extractor(self, method, args):
```
Comment (erikwijmans) [inline habitat_sim/utils/data/data_extractor.py:252]: We currently have a registry pattern implemented by `habitat_sim/registry.py`.  I suggest using that, that way users can easily ad their own pose extractor.

Comment (mpiseno) [reply habitat_sim/utils/data/data_extractor.py]: Wasn't 100% sure how the registry system worked, but I did my best to figure it out. Let me know if there's anything that jumps out that needs changing.

File: habitat_sim/utils/data/data_extractor.py [hunk 3]
```diff
@@ -9,12 +9,28 @@
...
 import habitat_sim
 import habitat_sim.bindings as hsim
+import habitat_sim.registry as registry
 from habitat_sim.agent import AgentState
 from habitat_sim.utils.common import quat_from_two_vectors
 from habitat_sim.utils.data.data_structures import ExtractorLRUCache
 from habitat_sim.utils.data.pose_extractor import PoseExtractor
 
 
+def make_pose_extractor(name: str) -> PoseExtractor:
```
Comment (erikwijmans) [inline habitat_sim/utils/data/data_extractor.py:19]: This doesn't construct the class, just gets it it seems.

File: habitat_sim/utils/data/data_extractor.py [hunk 4]
```diff
@@ -63,7 +79,7 @@ def __init__(
         if sum(split) != 100:
             raise Exception("Train/test split must sum to 100.")
 
-        assert extraction_method in ["closest", "panorama"]
+        # assert extraction_method in ["closest", "panorama"]
```
Comment (erikwijmans) [inline habitat_sim/utils/data/data_extractor.py:82]: Remove

**Pull request merged**